### PR TITLE
Ignore `search` command when calculating dependencies

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
* Add `search` to the generic command names
* Add `dependencies` flag to `create-content-graph` to create the graph with dependencies for better debugging.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/__main__.py [hunk 1]
```diff
@@ -2287,13 +2287,20 @@ def error_code(config, **kwargs):
 )
 @click.option('-ud', '--use-docker', is_flag=True, help="Use docker service to run the content graph")
 @click.option('-us', '--use-existing', is_flag=True, help="Use existing service", default=False)
+@click.option('-d', '--dependencies', is_flag=True, help="Whether include dependencies in graph", default=False)
```
Comment (tkatzir) [inline demisto_sdk/__main__.py:2290]:
```suggestion
@click.option('-d', '--dependencies', is_flag=True, help="Whether dependencies should be included in the graph", default=False)
```

**Review by tkatzir — APPROVED** [review]
Approved, with a comment on the English.

File: demisto_sdk/commands/common/constants.py [hunk 2]
```diff
@@ -1151,7 +1151,7 @@ class PB_Status:
 
 ENDPOINT_FLEXIBLE_REQUIRED_ARGS = ["ip", "id", "hostname"]
 
-GENERIC_COMMANDS_NAMES = BANG_COMMAND_NAMES.union({'send-mail', 'send-notification', 'cve-latest', 'cve-search'})
+GENERIC_COMMANDS_NAMES = BANG_COMMAND_NAMES.union({'send-mail', 'send-notification', 'cve-latest', 'cve-search', 'search'})
```
Comment (dantavori) [inline demisto_sdk/commands/common/constants.py:1154]: it will affect id-set existing flow, by design?

File: demisto_sdk/__main__.py [hunk 3]
```diff
@@ -2306,6 +2313,8 @@ def create_content_graph(use_docker: bool = False, use_existing: bool = False, o
         use_docker=use_docker,
     ) as content_graph_interface:
         create_content_graph_command(content_graph_interface)
+        if dependencies:
```
Comment (dantavori) [inline demisto_sdk/__main__.py:2316]: why is it needed?

Comment (ilaner) [reply demisto_sdk/__main__.py:2316]: Easier to debug, to able to create a full graph with dependencies in one command to figure out why the dependencies are calculated

File: demisto_sdk/commands/common/constants.py (hunk 2 above)
Comment (ilaner) [reply demisto_sdk/commands/common/constants.py:1154]: It seems like the `search` commands are not being collected in the dependencies, but I don't find where it's stated.
I think it's safe to add it to GENERIC_COMMAND_NAMES beacuse it's also a generic command and should be ignored.

**Pull request merged**